pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go

```diff
@@ -215,3 +228,56 @@ ExecStartPre=` + execStartPreConfig + `
 ExecStart=` + execStart),
 	}
 }
+
+func getFetchTokenScriptFile() (extensionsv1alpha1.File, error) {
+	var script bytes.Buffer
+	if err := tplFetchToken.Execute(&script, map[string]interface{}{
+		"pathCredentialsToken":  downloader.PathCredentialsToken,
+		"pathCredentialsServer": downloader.PathCredentialsServer,
+		"pathCredentialsCACert": downloader.PathCredentialsCACert,
+		"pathAuthToken":         PathAuthToken,
+		"dataKeyToken":          resourcesv1alpha1.DataKeyToken,
+		"secretName":            logging.PromtailTokenSecretName,
+	}); err != nil {
+		return extensionsv1alpha1.File{}, err
+	}
+
+	return extensionsv1alpha1.File{
+		Path:        PathFetchTokenScript,
+		Permissions: pointer.Int32(0744),
+		Content: extensionsv1alpha1.FileContent{
+			Inline: &extensionsv1alpha1.FileContentInline{
+				Encoding: "b64",
+				Data:     utils.EncodeBase64(script.Bytes()),
+			},
+		},
+	}, nil
+}
+
+func getFetchTokenScriptUnit(execStartPre, execStart string) extensionsv1alpha1.Unit {
+	unitContent := `[Unit]
+Description=promtail token fetcher
+After=` + downloader.UnitName + `
```

Comment:
The `promtail` unit gets auto-restarted after `5s` when it failed/exited, so it's not strictly needed. If the token was not yet downloaded then it restarts until it exists.
Also, this `fetch-token-script` unit is not running continuously but only every `5m` just for executing the script. I don't know the exact semantics, but couldn't it happen that `promtail` stays offline for at most `5m` if it fails intermittently?
I would suggest to proceed without this for now since there seems not to be a strong reason for doing it.

Reply:
thanks for the clarification. I agree that the restart 5s restart logic of the promtail unit will eventually lead to a working promtail. Let us leave it like it is.